### PR TITLE
Update produce block funcs to support Capella

### PR DESCRIPTION
**Pull request by saolyn**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

This PR updates the functions `produceBlockV2` and `produceBlindedBlock` to support Capella.
The following functions in `beacon-chain/rpc/eth/validator/validator.go` were modified:
   -  `ProduceBlockV2`
   -  `ProduceBlockV2SSZ`
   - `ProduceBlindedBlock`
   - `ProduceBlindedBlockSSZ`

### CODE REVIEW

File: beacon-chain/rpc/eth/validator/validator.go [hunk 1]
```diff
@@ -586,6 +634,21 @@ func (vs *Server) ProduceBlindedBlockSSZ(ctx context.Context, req *ethpbv1.Produ
 			Data:    sszBlock,
 		}, nil
 	}
+	capellaBlock, ok := v1alpha1resp.Block.(*ethpbalpha.GenericBeaconBlock_Capella)
+	if ok {
+		block, err := migration.V1Alpha1BeaconBlockCapellaToV2(capellaBlock.Capella)
```
Comment (saolyn) [inline beacon-chain/rpc/eth/validator/validator.go:639]: I am not sure this is correct, I could not find the Capella equivalent of `V1Alpha1BeaconBlockBellatrixToV2Blinded`

Comment (rkapka) [reply beacon-chain/rpc/eth/validator/validator.go:639]: You need to write this function. Fortunately I already have it here: https://gist.github.com/rkapka/222a4224be3ccf64808528927e74d881

File: beacon-chain/rpc/eth/validator/validator.go [hunk 2]
```diff
@@ -491,29 +519,49 @@ func (vs *Server) ProduceBlindedBlock(ctx context.Context, req *ethpbv1.ProduceB
...
-	}
-	if optimistic {
-		return nil, status.Errorf(codes.Unavailable, "The node is currently optimistic and cannot serve validators")
-	}
-	b, err := vs.V1Alpha1Server.GetBeaconBlock(ctx, v1alpha1req)
-	if err != nil {
-		return nil, status.Error(codes.Unavailable, "Could not get block from prysm API")
+	bellatrixBlock, ok := v1alpha1resp.Block.(*ethpbalpha.GenericBeaconBlock_BlindedBellatrix)
+	if ok {
+		// After Bellatrix, return blinded block.
```
Comment (rkapka) [inline beacon-chain/rpc/eth/validator/validator.go:524]: Let's remove this comment

File: beacon-chain/rpc/eth/validator/validator.go [hunk 3]
```diff
@@ -491,29 +519,49 @@ func (vs *Server) ProduceBlindedBlock(ctx context.Context, req *ethpbv1.ProduceB
...
+			Version: ethpbv2.Version_BELLATRIX,
+			Data: &ethpbv2.BlindedBeaconBlockContainer{
+				Block: &ethpbv2.BlindedBeaconBlockContainer_BellatrixBlock{BellatrixBlock: blk},
+			},
+		}, nil
 	}
-	blk, err := migration.V1Alpha1BeaconBlockBlindedBellatrixToV2Blinded(b.GetBlindedBellatrix())
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "Could not prepare beacon block: %v", err)
+	// Capella return blinded block
```
Comment (rkapka) [inline beacon-chain/rpc/eth/validator/validator.go:543]: Let's remove this comment

File: beacon-chain/rpc/eth/validator/validator.go [hunk 4]
```diff
@@ -491,29 +519,49 @@ func (vs *Server) ProduceBlindedBlock(ctx context.Context, req *ethpbv1.ProduceB
...
+	bellatrixBlock, ok := v1alpha1resp.Block.(*ethpbalpha.GenericBeaconBlock_BlindedBellatrix)
+	if ok {
+		// After Bellatrix, return blinded block.
+		optimistic, err := vs.OptimisticModeFetcher.IsOptimistic(ctx)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not determine if the node is a optimistic node: %v", err)
+		}
+		if optimistic {
+			return nil, status.Errorf(codes.Unavailable, "The node is currently optimistic and cannot serve validators")
+		}
```
Comment (rkapka) [inline beacon-chain/rpc/eth/validator/validator.go:531]: This same code is duplicated for Capella. We can move this check before we start processing bellatrix and do it only once.

**Review by rkapka — CHANGES_REQUESTED** [review]
Please add tests

**Review by rkapka — APPROVED** [review]

**Pull request merged**